edb/server/main.py

```diff
@@ -273,14 +274,24 @@ def run_server(args: ServerConfig):
 
     cluster: Union[pgcluster.Cluster, pgcluster.RemoteCluster]
     if args.data_dir:
-        if not args.max_backend_connections:
+        pg_max_connections = args.max_backend_connections
+        if not pg_max_connections:
             max_conns = _compute_default_max_backend_connections()
+            pg_max_connections = max_conns
```

Comment:
Please add a comment here on why we're doing this (the text of the commit message is perfect).